sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/implementation/StorageAccountImpl.java

```diff
@@ -637,4 +696,85 @@ public StorageAccountImpl withHnsEnabled(boolean enabled) {
         this.createParameters.withIsHnsEnabled(enabled);
         return this;
     }
+
+    private static final class PrivateLinkResourceImpl implements PrivateLinkResource {
+        private final com.azure.resourcemanager.storage.models.PrivateLinkResource innerModel;
+
+        private PrivateLinkResourceImpl(com.azure.resourcemanager.storage.models.PrivateLinkResource innerModel) {
+            this.innerModel = innerModel;
+        }
+
+        @Override
+        public String groupId() {
+            return innerModel.groupId();
+        }
+
+        @Override
+        public List<String> requiredMemberNames() {
+            return Collections.unmodifiableList(innerModel.requiredMembers());
+        }
+
+        @Override
+        public List<String> requiredDnsZoneNames() {
+            return Collections.unmodifiableList(innerModel.requiredZoneNames());
+        }
+    }
+
+    private static final class PrivateEndpointConnectionImpl implements PrivateEndpointConnection {
+        private final PrivateEndpointConnectionInner innerModel;
+
+        private final PrivateEndpoint privateEndpoint;
+        private final PrivateLinkServiceConnectionState privateLinkServiceConnectionState;
+        private final PrivateEndpointConnectionProvisioningState provisioningState;
+
+        public PrivateEndpointConnectionImpl(PrivateEndpointConnectionInner innerModel) {
```

Comment:
Wrapper code.

Reply:
Is it possible to put into resources for common usage?

Reply:
For common impl class, it would need some interface in inner class (like `ProxyResource` `Resource` `SupportsXXX`), which currently it not here.

Could consider adding it to codegen. Though current about 20 lines * about 5 services is not too much.